base/dSFMT.jl

```diff
@@ -79,20 +81,94 @@ function dsfmt_fill_array_close_open!(s::DSFMT_state, A::Ptr{Float64}, n::Int)
           s.val, A, n)
 end
 
+# dSFMT jump poly calculation
+
+"Represents a polynomial in GF(2)[X]"
+immutable GF2X
+    z::BigInt
+end
+
+GF2X(s::AbstractString) = GF2X(parse(BigInt, reverse(s), 16))
+Base.string(f::GF2X) = reverse(base(16, f.z))
+Base.:(==)(f::GF2X, g::GF2X) = f.z == g.z
+Base.copy(f::GF2X) = GF2X(f.z+0)
+
+degree(f::GF2X) = Base.ndigits0z(f.z, 2) - 1
+coeff(f::GF2X, i) = tstbit(f.z, i)
+setcoeff!(f::GF2X, i) = (setbit!(f.z, i); g)
+
+tstbit(z::BigInt, i) = ccall((:__gmpz_tstbit, :libgmp), Cint, (Ptr{BigInt}, Culong), &z, i) % Bool
+setbit!(z::BigInt, i) = ccall((:__gmpz_setbit, :libgmp), Void, (Ptr{BigInt}, Culong), &z, i)
+mulx!(f::GF2X, c::Int=1) = (ccall((:__gmpz_mul_2exp, :libgmp), Void, (Ptr{BigInt}, Ptr{BigInt}, Culong), &f.z, &f.z, c); f)
+xor!(f::GF2X, g::GF2X) = (ccall((:__gmpz_xor, :libgmp), Void, (Ptr{BigInt}, Ptr{BigInt}, Ptr{BigInt}), &f.z, &f.z, &g.z); f)
+
+# compute f*X mod m
+function mulxmod!(f::GF2X, m::GF2X, deg=degree(m))::GF2X
+    # precondition: degree(f) < degree(m)
+    mulx!(f)
+    degree(f) == deg && xor!(f, m)
+    f
+end
+
+# cache for X^(2i) mod m
+const _squares = Dict{GF2X, Vector{GF2X}}()
+
+# compute f^2 mod m
+function sqrmod!(f::GF2X, m::GF2X)::GF2X
+    d = degree(m)-1
+    0 <= degree(f) <= d || throw(DomainError())
+    sqrs = get!(_squares, m) do
+        x2i = GF2X(1)
+        GF2X[copy(mulxmod!(mulxmod!(x2i, m, d+1), m, d+1)) for i=1:d]
+    end
+    foldl(GF2X(0), filter(i->coeff(f, i), 0:degree(f))) do g, i
+        i <= d÷2 ? # optimization for "simple" squares
+            setcoeff!(g, 2i) :
+            xor!(g, sqrs[i])
+    end
+end
+
+# compute X^e mod m
+function powxmod(e::BigInt, m::GF2X)::GF2X
+    e < 0 && throw(DomainError())
+    foldl(GF2X(1), Base.ndigits0z(e, 2)-1:-1:0) do f, i
+        tstbit(e, i) ?
+            mulxmod!(sqrmod!(f, m), m) :
+            sqrmod!(f, m)
+    end
+end
+
+"Cached jump polynomials for `MersenneTwister`."
+# hack: contains Poly19937 at key -1 (as it can not be initialized at compile time)
```

Comment:
I just checked and it works! but I'll move anyway the comment after the definition.